### PR TITLE
STAC-20176: Additional metrics for the connection check

### PR DESCRIPTION
**Pull request by craffit**
### Step 3: Did you add / update tests for your changes in the right area?
- [ ] Unit/Component test		

### Step 4: I'm confident that everything is properly tested:
I got a PO / QA Approval by:
- [ ] Name

### Step 5: Can we ship this feature to production?
- [ ] Yes, I'm proud of my work. Ship it! :ship:

### CODE REVIEW

File: checks/net.go [hunk 1]
```diff
@@ -173,11 +173,32 @@ func (cp *connectionsPodsIndex) addPodWithPID(pod *kubelet.Pod, pid int32) {
...
+	}, []string{"result"})
+
+	httpMetricProcessedCounter = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "stackstate_process_agent",
+		Subsystem: "http_metric",
+		Name:      "processed",
+		Help:      "Http metrics processed by the connections check and the processing result",
+	}, []string{"result"})
+
+	httpObservationProcessedCounter = promauto.NewGaugeVec(prometheus.GaugeOpts{
```
Comment (aacevedoosorio) [inline checks/net.go:190]: httpObservationProcessedGauge

File: checks/net.go [hunk 2]
```diff
@@ -173,11 +173,32 @@ func (cp *connectionsPodsIndex) addPodWithPID(pod *kubelet.Pod, pid int32) {
...
-	connectionCounter = promauto.NewCounterVec(prometheus.CounterOpts{
+	connectionProcessedCounter = promauto.NewCounterVec(prometheus.CounterOpts{
 		Namespace: "stackstate_process_agent",
 		Subsystem: "connections",
 		Name:      "processed",
-		Help:      "Number of connections processed by the connections check",
+		Help:      "Connections processed by the connections check and the processing result",
+	}, []string{"result"})
+
+	httpMetricProcessedCounter = promauto.NewGaugeVec(prometheus.GaugeOpts{
```
Comment (aacevedoosorio) [inline checks/net.go:183]: httpMetricProcessedGauge

File: checks/net.go [hunk 3]
```diff
@@ -203,52 +255,92 @@ func (c *ConnectionsCheck) formatConnections(
...
+			},
+		})
+
 		// Check to see if this is a process that we observed and that it's not short-lived / blacklisted in the Process check
 		process, ok := processes[conn.Pid]
 		if !ok {
-			connectionCounter.WithLabelValues("no_process").Inc()
+			httpMetricNoProcess += httpMetricsCount
+			httpObservationNoProcess += httpObservationsCount
+			connectionProcessedCounter.WithLabelValues("no_process").Inc()
```
Comment (aacevedoosorio) [inline checks/net.go:325]: is the intention to have this metric with `no_process` label continuously growing while the same metric with a different level reflects a per run value?

File: checks/net.go [hunk 4]
```diff
@@ -203,52 +255,92 @@ func (c *ConnectionsCheck) formatConnections(
...
 		// put it in the cache for the next run
 		c.cache.PutNetworkRelationCache(relationID)
 
 		if cfg.EnableShortLivedNetworkRelationFilter &&
 			(!ok || isRelationShortLived(relationCache.FirstObserved, cfg)) {
-
-			connectionCounter.WithLabelValues("short_living").Inc()
+			httpMetricShortLived += httpMetricsCount
+			httpObservationShortLived += httpObservationsCount
+			connectionProcessedCounter.WithLabelValues("short_living").Inc()
```
Comment (aacevedoosorio) [inline checks/net.go:343]: Same question about a Counter or Gauge treatment?

File: checks/net.go (hunk 1 above)
Comment (craffit) [reply checks/net.go:190]: fixed

File: checks/net.go (hunk 2 above)
Comment (craffit) [reply checks/net.go]: fixed

File: checks/net.go (hunk 3 above)
Comment (craffit) [reply checks/net.go]: good point, making this a gauge aswell

File: checks/net.go (hunk 4 above)
Comment (craffit) [reply checks/net.go]: see above

**Review by aacevedoosorio — APPROVED** [review]

**Pull request merged**